src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala

```diff
@@ -512,4 +515,22 @@ final case class GatewayApiServiceImpl(
    */
   private def extractCategoryIpa(attributes: Seq[AttributeManagementApiAttribute]): String =
     attributes.find(_.origin == "IPA".some).map(_.name).getOrElse("Unknown")
+
+  override def getKeyJWKfromKId(kId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerKey: ToEntityMarshaller[Key],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize {
+    val operationLabel = s"Retrieving JWK of key with kId: $kId"
+    logger.info(operationLabel)
+
+    val result: Future[Key] = for {
+      maybeKey <- readModel.findOne[Key]("keys", Filters.eq("data.kId", kId))
```

Comment:
Queries in mongo are case sensitive (let's remember to save this field in lower case)
```suggestion
      maybeKey <- readModel.findOne[Key]("keys", Filters.eq("data.kid", kId))
```